### PR TITLE
Skip csrf on json.

### PR DESCRIPTION
**Pull request by jmcarp**
Unless there's a better way to do this--IANARD!

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,11 +1,9 @@
 class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
-  protect_from_forgery with: :exception
+  protect_from_forgery with: :exception unless -> { request.format.json? }
```
Comment (amoose) [inline app/controllers/application_controller.rb]: This is a fine start, but needs to be localized to the messages API.

**Comment by jmcarp** [comment]
@amoose: do you mean that other controllers would want to enable csrf checks for json requests? If so, why?

**Comment by jmcarp** [comment]
@amoose: added tests. I think skipping csrf checks is probably the right behavior for all json requests, so I didn't look into restricting that logic to messages--wdyt?

**Comment by amoose** [comment]
THANK YOU @jmcarp! Closing in favor of new PR #89

**Comment by amoose** [comment]
#89 has been tested and deployed. Going with that! 👍

**Pull request closed without merging**